Cockpit 289 (#624)
cockpit-project/cockpit-project.github.io: _posts/2023-04-05-cockpit-289.md

---
title: Cockpit 289
author: garrett
date: '2023-04-05'
tags: cockpit, machines, podman
slug: cockpit-289
category: release
summary: ''
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
We release regularly.

Here are the release notes from Cockpit 289, cockpit-machines 287, and cockpit-podman 66:

> ## Metrics: new looks with colorblind friendy colors for graphs

Author: ```suggestion
## Metrics: Indicate extreme usage and use colorblind-friendly colors
```

Author: ```suggestion
## Metrics: Indicate high usage and use colorblind-friendly colors
```

![228601347-df99586b-3c21-467f-a74d-1c0f020c2afc]({{ site.baseurl }}/images/289-metrics-new-looks-with-colorblind-friendy-colors-for-graphs.png)

> ## Accounts: Improve password setting

Author: ```suggestion
## Accounts: Improve password validation
```

State: **merged** @ <https://github.com/cockpit-project/cockpit/pull/18520>
{:.note}

There are several improvements to the process of setting password during account creation:
- The confirmation password is validated and doesn't match, validation is done after each keystroke. This improves user experience, as the user doesn't have to click "Create" whenever they want to know if passwords match.
- Password validation is now presented as warning, since a weak password is not actually forbidden, but discouraged nevertheless.

![Screenshot from 2023-04-05 12-41-47]({{ site.baseurl }}/images/289-accounts-improve-password-setting.png)

> ## Machines: Show VM's pending changes

Author: ```suggestion
## Machines: Show "pending changes" when configuration changes require a shutdown to take effect
```

State: **OPEN** (will not be included) @ <https://github.com/cockpit-project/cockpit-machines/pull/998>
{:.warn}

Add label which indicated whetever VM has pending changes and needs shutdown, and shows what specific configuration changes cause it.

> ## Machines: if virtualization is disabled in BIOS/EFI we now show message at initial load

Reviewer: Reviewing heading:

Machines: show an alert when virtualization is disabled in BIOS/EFI

State: **merged** @ <https://github.com/cockpit-project/cockpit-machines/pull/980>
{:.note}

![screenshot of if virtualization is disabled in bios/efi we now show message at initial load]({{ site.baseurl }}/images/289-machines-if-virtualization-is-disabled-in-bios-efi-we-now-show-message-at-initial-load.png)

Thanks to Subho Ghosh for this contribution.

## Machines: VSOCK interface

State: **OPEN** (will not be included) @ <https://github.com/cockpit-project/cockpit-machines/pull/969>
{:.warn}

> ## If storage file fails to get deleted, the disk will still get detached from the VM, but we show a warning:

Author: Oh, is this a feature that is supposed to be included also?

Why are there two here that don't have a URL in preview mode? Odd.

I can't find it. 

This looks related, perhaps? https://github.com/cockpit-project/cockpit-machines/pull/857

Author: It has a needs-work and also release-note. Since it needs-work, I guess it's not landing in this release?

Reviewer: Not going in in this release.

![Screenshot from 2022-11-14 18-48-54]({{ site.baseurl }}/images/289-if-storage-file-fails-to-get-deleted-the-disk-will-still-get-detached-from-the-vm-but-we-show-a-warning-.png)

> ## Prune unused containers support

Reviewer: Podman: support pruning unused containers?

State: **OPEN** (will not be included) @ <https://github.com/cockpit-project/cockpit-podman/pull/1235>
{:.warn}

<fill in>

![screenshot of ## prune unused containers support]({{ site.baseurl }}/images/289-prune-unused-containers-support.png)

> ## Specify a custom healthcheck action

Author: ```suggestion
## Podman: Custom healthcheck actions
```

State: **merged** @ <https://github.com/cockpit-project/cockpit-podman/pull/1233>
{:.note}

When a container becomes unhealthy a custom action can be configured in Podman, such as restarting the container. This requires Podman 4.3 or higher.

![screenshot of ## specify a custom healthcheck action]({{ site.baseurl }}/images/289-specify-a-custom-healthcheck-action.png)

> ## Sort containers

Reviewer: Podman: sort containers

State: **merged** @ <https://github.com/cockpit-project/cockpit-podman/pull/1232>
{:.note}

The podman container section now allows a user to sort on name, owner, CPU, memory and state.

![screenshot of ## sort containers]({{ site.baseurl }}/images/289-sort-containers.png)

> ## Podman: Explain the difference between a user and system session

Author: ```suggestion
## Podman: Explain the difference between user and system sessions
```

> ## Podman: Explain the difference between user and system sessions

Reviewer: I think that's not release note worthy.

State: **merged** @ <https://github.com/cockpit-project/cockpit-podman/pull/1227>
{:.note}

When creating a container with the system session offer a popover with the differences between the owner types.

Closes: #1067

---

![screenshot of explain the difference between a user and system session]({{ site.baseurl }}/images/289-podman-explain-the-difference-between-a-user-and-system-session.png)

> ## Development: Cockpit now supports the esbuild bundler

Reviewer: This can be removed as it's from previous release. My fault not removing the label sorry.
